### PR TITLE
[Cardano] Support utxo with legacy(byron) address

### PR DESCRIPTION
**Pull request by Balashov152**
## Description

In my project, I have old users with unspent Byron outputs. I want to support this case.

## How to test

1. You must have unspent output with the **byron** address on your wallet.
2. Just try to spend them. The transaction must be successfully sent

## Types of changes

- Add support to send legacy utxos in Cardano.
- Added an export method to work with a legacy address.

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: swift/Tests/Blockchains/CardanoTests.swift [hunk 1]
```diff
@@ -62,6 +62,51 @@ class CardanoTests: XCTestCase {
...
+
+        let encoded = output.encoded
+        XCTAssertEqual(encoded.hexString,
+            "83a40082825820554f2fd942a23d06835d26bbd78f0106fa94c8a551114a0bef81927f66467af000825820f074134aabbfb13b8aec7cf5465b1e5a862bde5cb88532cc7e64619179b3e76701018282583901558dd902616f5cd01edcc62870cb4748c45403f1228218bee5b628b526f0ca9e7a2c04d548fbd6ce86f358be139fe680652536437d1d6fd51a006acfc082583901df58ee97ce7a46cd8bdeec4e5f3a03297eb197825ed5681191110804df22424b6880b39e4bac8c58de9fe6d23d79aaf44756389d827aa09b1a000c9181021a0002b0bf031a032dcd55a20081825820a98fa31516d17fff3724e0d28b2b68f121a167cbf2205c4bddbe4acba265725b5840f82a6fd89ce3618a63f54ab9aeebc55bd3adea3a1c9c7ba4a97f3c69773c16cf6c0cee0a68d91eb318ca821a57b3a9e1a3c378d06b8ded98d9fee10307bf8d0e0281845820a98fa31516d17fff3724e0d28b2b68f121a167cbf2205c4bddbe4acba265725b5840f82a6fd89ce3618a63f54ab9aeebc55bd3adea3a1c9c7ba4a97f3c69773c16cf6c0cee0a68d91eb318ca821a57b3a9e1a3c378d06b8ded98d9fee10307bf8d0e5820000000000000000000000000000000000000000000000000000000000000000041a0f6")
+
+        let txid = output.txID
+        XCTAssertEqual(txid.hexString, "c83a6aec9c97a8c9f17e5b8e921c79f84ee150f062d1aa3cd0baaf60a48f790e")
```
Comment (satoshiotomakan) [inline swift/Tests/Blockchains/CardanoTests.swift:108]: Is there a successfully broadcasted transaction? I can't find this one: https://explorer.cardano.org/en/transaction?id=c83a6aec9c97a8c9f17e5b8e921c79f84ee150f062d1aa3cd0baaf60a48f790e

**Review by satoshiotomakan — APPROVED** [review]
LGTM!

File: swift/Tests/Blockchains/CardanoTests.swift (hunk 1 above)
Comment (Balashov152) [reply swift/Tests/Blockchains/CardanoTests.swift:108]: No, it's just a testing transaction. I will change it with a existing transition in a few hours.

**Review by lamafab — APPROVED** [review]

File: tests/chains/Cardano/SigningTests.cpp [hunk 2]
```diff
@@ -598,40 +598,43 @@ TEST(CardanoSigning, SignTransfer_0db1ea) {
     EXPECT_EQ(hex(txid), "0db1ea8c5c5828bbd027fcef3da02a63b86899db670ad7bb0630cefbe35944fa");
 }
 
+/// Successfully broadcasted:
+/// https://cardanoscan.io/transaction/87ca43a36b09c0b140f0ef2b71fbdcfcf1fdc88f7aa378b861e8eed3e8974628
```
Comment (Milerius) [inline tests/chains/Cardano/SigningTests.cpp:602]: tx id in the test if different than the link of this tx

**Review by Milerius — CHANGES_REQUESTED** [review]
Also missing android unit test

File: tests/chains/Cardano/SigningTests.cpp (hunk 2 above)
Comment (Balashov152) [reply tests/chains/Cardano/SigningTests.cpp]: Fixed

**Comment by Balashov152** [comment]
> Also missing android unit test

Added a test

**Comment by Balashov152** [comment]
Fixed android test

**Review by Milerius — APPROVED** [review]

**Review by satoshiotomakan — APPROVED** [review]

**Pull request merged**